pkg/services/ngalert/provisioning/notification_policies.go

```diff
@@ -7,22 +7,25 @@ import (
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/setting"
 )
 
 type NotificationPolicyService struct {
 	amStore         AMConfigStore
 	provenanceStore ProvisioningStore
 	xact            TransactionManager
 	log             log.Logger
+	settings        *setting.Cfg
```

Comment:
I avoided the config itself because we store the DefaultConfiguration as a raw string and not a struct, so anything could be passed in. IMO this approach was more strongly typed.

I went with a middle ground and only passed in the UnifiedAlerting settings. This is more scoped down, but style more type safe than `string`.